lib/oga/xml/sax_parser.rb

```diff
@@ -76,6 +76,10 @@ def initialize(handler, *args)
 
       # Delegate all callbacks to the handler object.
       instance_methods.grep(/^(on_|after_)/).each do |method|
+        if [:on_element, :after_element, :on_attribute, :on_attributes, :on_text].include?(method)
```

Comment:
Should I shorten this line? Maybe use `%i`?

Reply:
If you move this code to the end of this class (so explicit overrides already exist) I think you can just do this instead:

```ruby
existing_methods = instance_methods(false)

instance_methods.grep(/.../).each do |method|
  next if existing_methods.include?(method)

  eval ...
end
```

This would remove the need for explicitly tracking which methods should be ignored.

Reply:
Good idea, and it indeed works!